### PR TITLE
Event#update script bugfix

### PR DESCRIPTION
**Pull request by kunagpal**
`Event#update` now works correctly with the following script types:

* `Strings`
* `Arrays`
* `Script` instances
* Plain script objects (for backward compatibility)

### CODE REVIEW

File: lib/collection/event.js [hunk 1]
```diff
@@ -56,7 +56,10 @@ _.assign(Event.prototype, /** @lends Event.prototype */ {
              * The script that is to be executed when this event is triggered.
              * @type {Script}
              */
-            script: _.isString(definition.script) ? definition.script : new Script(definition.script)
+            script: Script.isScript(definition.script) ? definition.script :
```
Comment (czardoz) [inline lib/collection/event.js]: Too much ternary

File: test/unit/event.test.js [hunk 2]
```diff
@@ -1,18 +1,60 @@
...
+            var event = new Event(rawEvent),
+                jsonified = event.toJSON();
+
             expect(jsonified.listen).to.eql(event.listen);
             // Script property checking is done independently
             expect(jsonified).to.have.property('script');
         });
     });
+
+    describe('udpate', function () {
```
Comment (czardoz) [inline test/unit/event.test.js]: typo

File: lib/collection/event-list.js [hunk 3]
```diff
@@ -40,7 +40,7 @@ _.assign(EventList.prototype, /** @lends EventList.prototype */ {
             // array we are compiling for return
             (parent !== this.__parent) && EventList.isEventList(parent.events) &&
                 (parentEvents = parent.events.listenersOwn(name)) && parentEvents.length &&
-                all.push.apply(all, parentEvents);
+                all.push.apply(all, parentEvents); // eslint-disable-line prefer-spread
```
Comment (czardoz) [inline lib/collection/event-list.js]: I can go ahead with the merge for now, but need to know why this was not causing problems before.

Comment (kunagpal) [inline lib/collection/event-list.js]: I'll put my best people on the case. 🔎

Comment (czardoz) [inline lib/collection/event-list.js]: That's not very encouraging

**Pull request merged**